packages/hash/api/src/graphql/typeDefs/comment.typedef.ts

```diff
@@ -0,0 +1,99 @@
+import { gql } from "apollo-server-express";
+
+export const commentTypedef = gql`
+  type Comment implements Entity {
+    properties: CommentProperties!
+
+    contents: [TextToken!]!
+
+    owner: User!
+
+    parent: Block!
+
+    # ENTITY INTERFACE FIELDS BEGIN #
+    """
+    The id of the entity - alias of 'entityId'
+    """
+    id: ID!
+    """
+    The id of the entity - alias of 'id'
+    """
+    entityId: ID!
+    """
+    The specific version if of the entity
+    """
+    entityVersionId: ID!
+    """
+    The id of the account this entity belongs to
+    """
+    accountId: ID!
+    """
+    The date the entity was created
+    """
+    createdAt: Date!
+    """
+    The date this entity version was created.
+    """
+    entityVersionCreatedAt: Date!
+    """
+    The user who created the entity
+    """
+    createdByAccountId: ID!
+    """
+    The date the entity was last updated
+    """
+    updatedAt: Date!
+    """
+    The visibility level of the entity
+    """
+    visibility: Visibility!
+    """
+    The fixed id of the type this entity is of
+    """
+    entityTypeId: ID!
+    """
+    The id of the specific version of the type this entity is of
+    """
+    entityTypeVersionId: ID!
+    """
+    The name of the entity type this belongs to.
+    N.B. Type names are unique by account - not globally.
+    """
+    entityTypeName: String!
+    """
+    The full entityType definition
+    """
+    entityType: EntityType!
+    """
+    The version timeline of the entity.
+    """
+    history: [EntityVersion!]
+    """
+    The outgoing links of the entity.
+    """
+    linkGroups: [LinkGroup!]!
+    """
+    The linked entities of the entity.
+    """
+    linkedEntities: [UnknownEntity!]!
+    """
+    The linked aggregations of the entity.
+    """
+    linkedAggregations: [LinkedAggregation!]!
+    # ENTITY INTERFACE FIELDS END #
+  }
+
+  type CommentProperties {
+    createdAt: Date!
+    resolvedAt: Date
+    editedAt: Date
+  }
```

Comment:
**Blocking:** why are there dedicated `createdAt` and `editedAt` fields in the `properties` of this entity? How are they different from the existing `entity.createdAt` and `entity.updatedAt` fields?

Is it because the updates to the linked text entities aren't currently propagated to this entity? In which case wouldn't a GQL field resolver be more suited for these, so that they don't have to be manually updated?

If there's no difference, I think they should be removed entirely.

Reply:
Adding `createdAt` was advised by @CiaranMn, because of possible discrepancies between DB insertion timestamp vs actual comment creation timestamp. I followed the same pattern for `updatedAt` for consistency.

Reply:
And is there a reason we can't use a field resolver, to avoid the frontend from having to manually set these fields?

Reply:
I think I misunderstood, you're saying that we want the timestamp to reflect when the user created the comment, rather than when the entity was created in the datastore. I don't understand this, because the difference should be a few milliseconds. Are you able to expand on the reasoning? I think it's worthwhile exploring because these properties are going to significantly burden the implementation.

Reply:
> I think one of the reasons for storing the timestamp in the properties instead of the entity was when migrating databases we could lose the creation timestamp of the comment (if the entity was reinserted into the db).

Properly implemented database migrations should preserve the `createdAt` timestamps of entities, so I'm not sure this is a great reason.

The only purpose these fields serve that I can think of is that they can reflect when the text entity has changed rather than the comment entity itself. However this can (and I think should) be implemented using a field resolver, as it is just an alias for data we are already capturing.